src/main/java/bio/terra/cloudfunctions/common/GsonWrapper.java

```diff
@@ -4,40 +4,73 @@
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonSyntaxException;
 import java.time.OffsetDateTime;
 import java.time.ZonedDateTime;
+import java.util.Map;
 
 public class GsonWrapper {
-  private static Gson _instance = initInstance();
+  private GsonWrapper() {}
+
+  // Provide a thread-safe Gson singleton (Bill Pugh)
+  private static class GsonSingleton {
+    private static final Gson _instance =
+        new GsonBuilder()
+            .setFieldNamingPolicy(FieldNamingPolicy.IDENTITY)
+            .registerTypeAdapter(
+                OffsetDateTime.class,
+                (JsonDeserializer<OffsetDateTime>)
+                    (json, type, jsonDeserializationContext) -> {
+                      try {
+                        return ZonedDateTime.parse(json.getAsString()).toOffsetDateTime();
+                      } catch (Exception e) {
+                        return null;
+                      }
+                    })
+            .create();
+  }
   /**
    * Returns a Gson instance capable of parsing a string that represents OffsetDateTime object.
    *
    * @return Gson object
    */
-  private static Gson initInstance() {
-    if (_instance == null)
-      _instance =
-          new GsonBuilder()
-              .setFieldNamingPolicy(FieldNamingPolicy.IDENTITY)
-              .registerTypeAdapter(
-                  OffsetDateTime.class,
-                  (JsonDeserializer<OffsetDateTime>)
-                      (json, type, jsonDeserializationContext) -> {
-                        try {
-                          return ZonedDateTime.parse(json.getAsString()).toOffsetDateTime();
-                        } catch (Exception e) {
-                          return null;
-                        }
-                      })
-              .create();
-    return _instance;
-  }
-
   public static Gson getInstance() {
-    return _instance;
+    return GsonSingleton._instance;
   }
 
-  public static <T> T convertFromClass(String s, Class<T> classOfT) {
+  /**
+   * @param s json string
+   * @param classOfT the class to cast the json string
+   * @param <T> the return type
+   * @return
+   * @throws JsonSyntaxException when something goes wrong
+   */
+  public static <T> T convertFromClass(String s, Class<T> classOfT) throws JsonSyntaxException {
     return getInstance().fromJson(s, classOfT);
   }
+
+  /**
+   * @param j an JsonElement object (e.g. JsonObject)
+   * @param classOfT the class to cast the json object
+   * @param <T> the return type
+   * @return
+   * @throws JsonSyntaxException when something goes wrong
+   */
+  public static <T> T convertFromClass(JsonElement j, Class<T> classOfT)
+      throws JsonSyntaxException {
+    return getInstance().fromJson(j, classOfT);
+  }
```

Comment:
The context that this method get called will always be a `JsonObject` since it was used by the following code but yes changing from `JsonElement` to `JsonObject` can safeguard the unscrupulous use of other element types.

`  public static <T> T convertFromClass(Map<?, ?> obj, Class<T> classOfT)
      throws JsonSyntaxException {
    return convertFromClass(getInstance().toJsonTree(obj).getAsJsonObject(), classOfT);
  }`